src/middlewared/middlewared/plugins/pool.py

```diff
@@ -3746,7 +3748,13 @@ async def kill_processes(self, oid, control_services, max_tries=5):
             if not processes:
                 return
 
+            mid_procs.extend([x.pid for x in psutil.Process(middleware_pid()).children()])
```

Comment:
We only need to keep main middleware process. If multiprocessing worker is killed, it's fine, it will be restarted. If rclone/rsync are killed, it's fine too.

Reply:
Okay. I'll only check for the main pid.